### ISSUE TITLE
bare literal types

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#46]
```py
foo: True
bar: 1 | 2
```

**Comment by KotlinIsland** [comment]
and `Enum`s

**Comment by anadodik** [comment]
This would be really nice for typing fixed-size arrays/vectors/matrices/etc. 

Similarly, supporting Ellipses `...` would be quite useful.

Is this on the roadmap?

**Comment by DetachHead** [comment]
adding non-standard syntax like this isn't super high priority for me, i consider fixing false positives/false negatives and language server features to be more important. however if someone wants to implement it a PR is welcome, as long as there's an option to ban it for users who either can't or don't want to use non-standard syntax in their projects

**Comment by KotlinIsland** [comment]
pyright already supports bare literals:
```py
class A[T]:
    def __init__(self, t: T) -> None:
        pass

B = A[1]
b: B = A(2)
```

**Comment by DetachHead** [comment]
that behavior is not intentional and is therefore completely broken

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=MYGwhgzhAECCDaAVAugLgFDS9AJgUwDNoB9YgSwDsyAXUgCgjxAIBpprVpEBKaAWgB80AHIB7CngzZp0AA6QI6dACFoAXjjwAjNAA%2B0AEzJ0AI06qNsOgd4BiaHgBOj0Y84AiStXfQyMCqLU0ApkAOYUYCYgeOyi7ACesjHuADI0TmAg8ADMyO5AA)

```python
class A[T]:
    def __init__(self, t: T) -> None:
        pass

B = A[1 | 2]
b: B = A(2) # error: "int" is not assignable to type "Literal[3]"
```
